rustls/src/quic.rs

```diff
@@ -32,6 +32,12 @@ impl Secrets {
 /// Generic methods for QUIC sessions
 pub trait QuicExt {
     /// Return the TLS-encoded transport parameters for the session's peer.
+    ///
+    /// While the transport parameters are technically available prior to the
+    /// completion of the handshake, they cannot be fully trusted until the
+    /// handshake completes, and reliance on them should be minimized.
+    /// However, any tampering with the parameters will cause the handshake
+    /// to fail.
     fn get_quic_transport_parameters(&self) -> Option<&[u8]>;
```

Comment:
We should file a separate issue for this: We should refactor this API to minimize the danger here.

Reply:
Yeah, I wanted to get @Ralith's eyes on it before changing it. While reviewing code in Quinn, it did seem like it relies on being able to access this early. (The change I had in mind was just to return `None` while `self.is_handshaking()`.)

Reply:
Responded in #655.